morpheus/_lib/src/messages/multi.cpp

```diff
@@ -284,13 +362,43 @@ void MultiMessageInterfaceProxy::set_meta(MultiMessage& self, pybind11::object c
 
     auto df = mutable_info.checkout_obj();
 
-    auto index_slice = pybind11::slice(
-        pybind11::int_(self.mess_offset), pybind11::int_(self.mess_offset + self.mess_count), pybind11::none());
+    auto [row_indexer, column_indexer] = get_indexers(self, df, columns);
+
+    // Check to see if this is adding a column. If so, we need to use .loc instead of .iloc
+    if (column_indexer.contains(-1))
+    {
+        // cudf is really bad at adding new columns. Need to use loc with a unique and monotonic index
+        py::object saved_index = py::none();
+
+        // Check to see if we can use slices
+        if (!df.attr("index").attr("is_unique").cast<bool>() || !df.attr("index").attr("is_monotonic").cast<bool>())
```

Comment:
I'm wondering if there would be any performance benefit to saving `df.attr("index")` to a variable like:
```
auto idx_attr = df.attr("index");
if (!idx_attr.attr("is_unique").cast<bool>() || !idx_attr.attr("is_monotonic").cast<bool>())
{
    saved_index = std::move(idx_attr);
```

Reply:
Fixed